### PR TITLE
add cve-2017-9772 to the 4.04 release page

### PR DESCRIPTION
**Pull request by avsm**

### CODE REVIEW

**Comment by avsm** [comment]
The site appears to need a custom opam2web to build, which is failing as the constraints need to include <js_of_ocaml.3.0.0 - cc @Chris00 

line in travis-ci.sh is:

```
opam pin add opam2web https://github.com/Chris00/opam2web.git#opl
```

File: site/releases/4.04.md [hunk 1]
```diff
@@ -20,6 +20,50 @@ bug fixes, optimizations, etc., please consult the
...
+However, in that situation, any user who execute the program gains all
+the privileges of the owner of the executable (meaning that root-owned
+setuid executables provide root access).
+
+Fix: OCaml 4.04.2 mitigates this by modifying Sys.getenv and Unix.getenv
+to raise an exception if the process has ever had elevated privileges.
+The OCaml runtime has also been modified to use this function for
+retrieving all of the runtime environment variables which could potentially
+cause files to be accessed or modified.  The older behaviour is available
+in Sys.unsafe_getenv for applications that require strict compatibility.
```
Comment (dra27) [inline site/releases/4.04.md]: This sentence isn't quite right - s/Sys.unsafe_getenv/the caml_sys_unsafe_getenv primitive/

File: site/releases/4.04.md [hunk 2]
```diff
@@ -20,6 +20,50 @@ bug fixes, optimizations, etc., please consult the
...
+
+Credits: This was originally reported by Eric Milliken on the OCaml Mantis
+bug tracker. https://caml.inria.fr/mantis/view.php?id=7557
+
+References: see CVE-2017-9779 for a lesser vulnerability in older versions.
```
Comment (dra27) [inline site/releases/4.04.md]: Have we dropped this one now?

**Comment by Chris00** [comment]
@avsm The custom `opam2web` is to have the [following patch applied](https://github.com/ocaml/opam2web/pull/150).

**Comment by Chris00** [comment]
What is the way to replace `js_of_ocaml` syntax now?  After all, it is only a small file (`search.ml`) to update.

**Comment by Chris00** [comment]
I required `js_of_ocaml-camlp4` in the opam file of `opam2web`.

**Comment by Chris00** [comment]
Build passing.  It would be good to answer @dra27 concerns before I merge this PR.

**Comment by Chris00** [comment]
P.S. On the ocaml.org server, I had to unpin then pin again `opam2web` for opam to synchronize with the updated version of the `#opl` branch...

**Comment by avsm** [comment]
@Chris00 I've cut a release of opam2web to ease the dependencies

File: site/releases/4.04.md (hunk 1 above)
Comment (avsm) [reply site/releases/4.04.md]: fixed infaf593f3c7d8640e94d2cd7ec9158e4cfb93a04b

File: site/releases/4.04.md (hunk 2 above)
Comment (avsm) [reply site/releases/4.04.md]: yep, removed in faf593f3c7d8640e94d2cd7ec9158e4cfb93a04b

**Comment by Chris00** [comment]
@avsm Thanks.  Ready to unpin it when the new package will be available.

**Pull request merged**